### PR TITLE
chore: [ANDROSDK-1891] Expose an alternative method for API interaction

### PR DESCRIPTION
**Pull request by taridepaco**

### CODE REVIEW

File: core/src/main/java/org/hisp/dhis/android/core/D2.kt [hunk 1]
```diff
@@ -74,8 +74,7 @@ import org.hisp.dhis.android.core.wipe.internal.WipeModule
 class D2 internal constructor(internal val d2DIComponent: D2DIComponent) {
     private val modules: D2Modules = d2DIComponent.modules
 
-    @VisibleForTesting
-    internal fun ktor(): KtorServiceClient {
+    fun ktor(): HttpServiceClient {
```
Comment (vgarciabnz) [inline core/src/main/java/org/hisp/dhis/android/core/D2.kt:77]: This method is exposed and is named `ktor`, shouldn't it be called `httpServiceClient`?

File: core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt [hunk 2]
```diff
@@ -44,11 +44,11 @@ import okhttp3.ResponseBody.Companion.toResponseBody
 import org.koin.core.annotation.Singleton
 
 @Singleton
-internal class KtorServiceClient(
-    private val client: HttpClient,
+class HttpServiceClient(
+    val client: HttpClient,
```
Comment (vgarciabnz) [inline core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt:48]: These two properties are now exposed in the public API. The `client` property is ktor-specific. Couldn't they be hidden by marking them as internal or private?

File: core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt [hunk 3]
```diff
@@ -44,11 +44,11 @@ import okhttp3.ResponseBody.Companion.toResponseBody
...
     var baseUrl: String = "https://temporary-dhis-url.org/api/",
 ) {
-    internal suspend inline fun <reified T> request(
+    suspend inline fun <reified T> request(
```
Comment (vgarciabnz) [inline core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt:51]: If this method is public, the parameter `requestMethod` is exposed, which exposes a ktor class `HttpMethod`. Couldn't it be hidden? Maybe the annotation PublishedApi can help here

**Review by vgarciabnz — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt (hunk 2 above)
Comment (taridepaco) [reply core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt:48]: It seems Public-API inline function cannot access non-public-API internal variables.

Comment (vgarciabnz) [reply core/src/main/java/org/hisp/dhis/android/core/arch/api/internal/HttpServiceClient.kt:48]: the publishedAPI annotation doesn't help here?

**Review by andresmr — APPROVED** [review]

**Review by vgarciabnz — APPROVED** [review]

**Pull request merged**